pkg/controller/csr_check.go

```diff
@@ -231,47 +249,63 @@ func authorizeCSR(
 }
 
-func authorizeNodeClientCSR(c client.Client, config ClusterMachineApproverConfig, machines []machinev1.Machine, req *certificatesv1.CertificateSigningRequest, csr *x509.CertificateRequest) error {
-	if config.NodeClientCert.Disabled {
-		return fmt.Errorf("CSR %s for node client cert rejected as the flow is disabled", req.Name)
-	}
+func authorizeNodeClientCSR(c client.Client, machines []machinev1.Machine, req *certificatesv1.CertificateSigningRequest, csr *x509.CertificateRequest) (bool, error) {
 
 	if !isReqFromNodeBootstrapper(req) {
-		return fmt.Errorf("CSR %s for node client cert has wrong user %s or groups %s", req.Name, req.Spec.Username, sets.NewString(req.Spec.Groups...))
+		klog.Infof("%v: CSR does not appear to be a valid node bootstrapper client cert request", req.Name)
+		return false, nil
 	}
 
 	nodeName := strings.TrimPrefix(csr.Subject.CommonName, nodeUserPrefix)
 	if len(nodeName) == 0 {
-		return fmt.Errorf("CSR %s has empty node name", req.Name)
+		//TODO: set annotation/emit event here.
+		klog.Errorf("%v: CSR does not appear to be a valid node bootstrapper client cert request", req.Name)
+		return false, nil
 	}
 
-	if err := c.Get(context.Background(), client.ObjectKey{Name: nodeName}, &corev1.Node{}); !apierrors.IsNotFound(err) {
-		return fmt.Errorf("node %s already exists or other error: %v", nodeName, err)
+	if err := c.Get(context.Background(), client.ObjectKey{Name: nodeName}, &corev1.Node{}); err != nil && !apierrors.IsNotFound(err) {
+		// possible transient API error, requeue
+		klog.Errorf("%v: unable to get node %s error: %v", req.Name, nodeName, err)
+		return false, fmt.Errorf("failed get existing nodes %s", nodeName)
+	} else if err == nil {
+		//TODO: set annotation/emit event here.
+		klog.Errorf("%v: node %s already exists, cannot approve", req.Name, nodeName)
+		return false, nil
 	}
 
 	nodeMachine, ok := findMatchingMachineFromInternalDNS(nodeName, machines)
 	if !ok {
-		return fmt.Errorf("failed to find machine for node %s", nodeName)
+		//TODO: set annotation/emit event here.
+		klog.Errorf("%v: failed to find machine for node %s, cannot approve", req.Name, nodeName)
+		return false, fmt.Errorf("failed to find machine for node %s", nodeName)
 	}
 
 	if nodeMachine.Status.NodeRef != nil {
-		return fmt.Errorf("machine for node %s already has node ref", nodeName)
+		//TODO: set annotation/emit event here.
+		klog.Errorf("%v: machine for node %s already has node ref, cannot approve", req.Name, nodeName)
+		return false, nil
 	}
 
 	start := nodeMachine.CreationTimestamp.Add(-maxMachineClockSkew)
 	end := nodeMachine.CreationTimestamp.Add(maxMachineDelta)
 	if !inTimeSpan(start, end, req.CreationTimestamp.Time) {
-		return fmt.Errorf("CSR %s creation time %s not in range (%s, %s)", req.Name, req.CreationTimestamp.Time, start, end)
+		//TODO: set annotation/emit event here.
+		klog.Errorf("%v: CSR creation time %s not in range (%s, %s)", req.Name, req.CreationTimestamp.Time, start, end)
```

Comment:
Could putting this timing information in here be used later to forge a valid CSR if these logs were somehow leaked?

Reply:
No.